doc/c-style-guide.md

```diff
@@ -0,0 +1,41 @@
+GRPC C STYLE GUIDE
+=====================
+
+Background
+----------
+
+Here we document style rules for C usage in the gRPC Core library.
+
+General
+-------
+
+- Layout rules are defined by clang-format, and all code should be passed through
+  clang-format. A (docker-based) script to do so is included in 
+  tools/distrib/clang_format_code.sh.
+
+Header Files
+------------
+
+- Public header files (those in the include/grpc tree) should compile as pedantic C89
```

Comment:
Nit: Isn't there a logical contradiction between "API not consumed directly by applications" and "extra bother for consumers of the API"? In any case, parts of the census API are intended to be used by applications. 

More importantly, the bool usage is only a (small) part of the justification here, I'm more skeptical that *anyone* is C89 only, on any platform of interest.